### PR TITLE
Load default translation

### PR DESCRIPTION
**Pull request by mentaljam**
Load default translation if locale-based translation haven't been loaded
(actual for id-based translations)

### CODE REVIEW

File: src/sailfishapp_priv.cpp [hunk 1]
```diff
@@ -108,8 +109,13 @@ configureApp(QGuiApplication *app)
         // Fallback translation (optional; only for id-based translations)
         //translator->load(QDir(translations).filePath(appName() + "_eng_en"));
 
-        // Locale-based translation
-        translator->load(QLocale::system(), appName(), "-", translations);
+        // Try to load locale-based translation
+        if (!translator->load(QLocale::system(), name, "-", translations))
+        {
+            // Load default translation if locale-based translation
+            // haven't been loaded (actual for id-based translations)
```
Comment (pvuorela) [inline src/sailfishapp_priv.cpp]: This and above commented out _eng_en loading are both for the same purpose. In addition I think QTranslator::load() fallbacks are already trying to use what is explicitly asked below. Cosmetic detail: inconsistent opening bracing compared to rest of the code.

Comment (mentaljam) [reply src/sailfishapp_priv.cpp]: > This and above commented out _eng_en loading are both for the same purpose.

So in this case a package must provide an `<app-name>_eng_en.qm` file but the SDK propose the `<appname>.qm` and `<appname>-<lang>.qm` files. So it seems to be a good idea to use the `<appname>.qm` file as default (in my opinion of course).

> In addition I think QTranslator::load() fallbacks are already trying to use what is explicitly asked below.

But I faced a problem of loading the default translation in my application so I had to load it explicitly after SailfishApp failed to load a translation for my current locale

Comment (pvuorela) [reply src/sailfishapp_priv.cpp]: File should get loaded like in http://doc.qt.io/qt-5/qtranslator.html#load-1

Checking QTranslator source, it does seem to omit file.qm if suffix is not explicitly given. Didn't test but adding ".qm" parameter to above load could already get your file loaded without this separate load().

Comment (mentaljam) [reply src/sailfishapp_priv.cpp]: > ...it does seem to omit file.qm if suffix is not explicitly given...

Oh, thank you! Well then I should test it in my app with explicitly given suffix.

Comment (mentaljam) [reply src/sailfishapp_priv.cpp]: Ok, I've added the `.qm` suffix to the load method and it's works on my device. I'm going to close this PR and create a new one with proper changes.

**Pull request closed without merging**